src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -280,7 +280,7 @@
     <value>The JSON value of length {0} is too large and not supported by the JSON writer.</value>
   </data>
   <data name="ZeroDepthAtEnd" xml:space="preserve">
-    <value>Expected CurrentDepth ({0}) to be zero at the end of the JSON payload. There is an open JSON object or array that should be closed.</value>
+    <value>Expected depth to be zero at the end of the JSON payload. There is an open JSON object or array that should be closed.</value>
```

Comment:
IMO this error should be something more like "Unexpected end of JSON payload."

Reply:
To clarify, without the `There is an open JSON object or array that should be closed.` qualifier?

Reply:
Hmm, what does XmlReader do? With Json.NET I have always used "unexpected end" as the exception message when in the middle of reading something and there is no more data. But I don't know if that is because I took the error message from other serializers or it is something I made up 😄

Reply:
Compilers most commonly say   "unexpected end of file" so I would expect developers to be familiar with it.